include/swift/Driver/ToolChain.h

```diff
@@ -83,6 +86,34 @@ class ToolChain {
     /// arguments.
     const char *getTemporaryFilePath(const llvm::Twine &name,
                                      StringRef suffix = "") const;
+
+    /// For frontend, merge-module, and link invocations.
+    bool shouldUseInputFileList() const;
+
+    bool shouldUsePrimaryInputFileListInFrontendInvocation() const;
+
+    bool shouldUseMainOutputFileListInFrontendInvocation() const;
+
+    bool shouldUseSupplementaryOutputFileMapInFrontendInvocation() const;
+
+    /// Reify the existing behavior that SingleCompile compile actions do not
+    /// filter, but batch-mode and single-file compilations do. Some clients are
+    /// relying on this (i.e., they pass inputs that don't have ".swift" as an
+    /// extension.) It would be nice to eliminate this distinction someday.
```

Comment:
Aha, this is probably the comment we were talking about! This form is fine, although it looks like an implementation comment rather than a declaration comment.